dashboard/src/shared/Auth.ts

```diff
@@ -76,6 +76,18 @@ export class Auth {
     }
   }
 
+  // is403FromAuthProxy returns true if the response is a 403 determined to have originated
+  // from the auth proxy itself, rather than upstream.
+  //
+  // Ideally we would be able to set a header for responses generated by the
+  // auth proxy, rather than rely on the fact that the 403 response sent by
+  // the auth proxy is (by default) an html page (rather than the json
+  // upstream result). Hence encapsulating this ugliness here so we can fix
+  // it in the one spot.
+  public static is403FromAuthProxy(r: AxiosResponse): boolean {
+    return r.status === 403 && (!r.data || !r.data.message);
```

Comment:
Yeah. It would be nice if oauth2_proxy allowed setting a header (given that we want to use a http-only cookie, so by definition can't test that). Using `/oauth2/userinfo` (or `/oauth2/auth` perhaps) would be nicer, though an extra request. I'll update the comment here to reflect that too.